### PR TITLE
Initial Implementation

### PR DESCRIPTION
**Pull request by gusnaughton**
# Alaska Airlines Pull Request

Introduces auro-flight. More to come

## Type of change:

Please delete options that are not relevant.

- [x] New capability
- [ ] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

**Comment by gusnaughton** [comment]
@erikalanghauser  indicated that this piece looks ready to go, pending a final review after https://github.com/AlaskaAirlines/auro-flightline/pull/2 is merged.

@blackfalcon there's a really strange issue with the rollup config not compiling for this repository that I'm having issues nailing down:
```
> @alaskaairux/auro-flight@0.0.0 bundler /Users/gusnaughton/git/auro-flight
> rollup -c


src/auro-flight.js → dist/auro-flight__bundled.js...
created dist/auro-flight__bundled.js in 1.7s

src/es5.js → dist/auro-flight__bundled.es5.js...
[!] Error: Could not resolve entry module (src/es5.js).
Error: Could not resolve entry module (src/es5.js).
    at error (/Users/gusnaughton/git/auro-flight/node_modules/rollup/dist/shared/rollup.js:5240:30)
    at ModuleLoader.loadEntryModule (/Users/gusnaughton/git/auro-flight/node_modules/rollup/dist/shared/rollup.js:18659:20)
    at async Promise.all (index 0)

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @alaskaairux/auro-flight@0.0.0 bundler: `rollup -c`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @alaskaairux/auro-flight@0.0.0 bundler script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/gusnaughton/.npm/_logs/2021-02-03T22_38_17_043Z-debug.log
ERROR: "bundler" exited with 1.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @alaskaairux/auro-flight@0.0.0 ciBuild: `npm-run-all sassBuild sassRender cssLint distJS bundler postinstall apiBuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @alaskaairux/auro-flight@0.0.0 ciBuild script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/gusnaughton/.npm/_logs/2021-02-03T22_38_17_086Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @alaskaairux/auro-flight@0.0.0 build: `npm run ciBuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @alaskaairux/auro-flight@0.0.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/gusnaughton/.npm/_logs/2021-02-03T22_38_17_124Z-debug.log
```

**Comment by geoffrich** [comment]
@gusnaughton you deleted the `src/es5.js` file. Rollup uses this as an entry point to bundle a legacy version of the component. See an example in the interruption repo: https://github.com/AlaskaAirlines/auro-interruption/blob/master/src/es5.js

**Comment by gusnaughton** [comment]
Resolved the rollup issue, was just missing src/es5.js with the new wc-generator update. @blackfalcon, Please review but do not merge.

**Comment by blackfalcon** [comment]
> Resolved the rollup issue, was just missing src/es5.js with the new wc-generator update. @blackfalcon, Please review but do not merge.

What does this mean? The file is in the install?
https://github.com/AlaskaAirlines/WC-Generator/blob/master/template/src/es5.js

**Review by blackfalcon — CHANGES_REQUESTED** [review]
Question: Why are these two components? I feel this adds overall complexity versus making this easier to use/build.

File: src/auro-flight-top-bar.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+      ${styleCss}
+    `;
+  }
+
+  /**
+   * Internal function to render either the flight number OR 'Multiple flights'.
+   *
+   * @returns {String} Parsed airline code output
+   */
+  generateFlights() {
```
[the remark this answers is not in the document]
Comment (gusnaughton) [reply src/auro-flight-top-bar.js:47]: It's a special setting I configured just for you

File: src/auro-flight-top-bar.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+   * 2+ day change = +N Days
+   *
+   * @returns {String} item to display
+   */
+  generateDays() {
+      if (this.daysChanged > ZERO) {
+          return `+${this.daysChanged} day${this.daysChanged > ONE ? 's' : ''}`
+      }
+
+return null;
```
[the remark this answers is not in the document]
Comment (gusnaughton) [reply src/auro-flight-top-bar.js:71]: I'm dynamic, deal with it.

**Review by erikalanghauser — DISMISSED** [review]
Looks fab! ✨

**Review by blackfalcon — APPROVED** [review]

File: demo/dotCompliance.md [hunk 3]
```diff
@@ -0,0 +1,376 @@
...
+        <auro-icon category="logos" name="tail-AS" style="width: 32px"></auro-icon>
+        AS 2087 is operated by Horizon Air as AlaskaHorizon
+      </span>
+
+  </auro-flight>
+  ```
+
+</auro-accordion>
+
+<!-- ## Flight under 1 hour / Shuttle Market (SEAPDX)
```
[the remark this answers is not in the document]
Comment (blackfalcon) [reply demo/dotCompliance.md]: Removed content. It's in the commit log if the demo is necessary to be brought back.

File: demo/dotCompliance.md [hunk 4]
```diff
@@ -0,0 +1,376 @@
...
+      </auro-flightline>
+      <span slot="footer"><auro-icon category="logos" name="tail-AA" style="width: 32px"></auro-icon>AA 3893 is operated by Envoy Air on behalf of American Airlines: ORD - AVP</span>
+    </auro-flight>
+  ```
+
+</auro-accordion>
+
+## Codeshare + Mainline + Regional Connection (DXBALW)
+
+**Why is this example important?**
```
[the remark this answers is not in the document]
Comment (blackfalcon) [reply demo/dotCompliance.md]: Removed content. It's in the commit log if the demo is necessary to be brought back.

File: demo/dotCompliance.md [hunk 5]
```diff
@@ -0,0 +1,376 @@
...
+        <auro-icon category="logos" name="tail-AS" style="width: 32px"></auro-icon>AS 1085 is operated by Alaska Airlines: LAX - SEA<br />
+        <auro-icon category="logos" name="tail-AS" style="width: 32px"></auro-icon>AS 2086 is operated by Horizon Air as AlaskaHorizon: SEA - ALW
+      </span>
+    </auro-flight>
+  ```
+</auro-accordion>
+
+## Single Regional, Multiple Partner (ALWCPT)
+
+**Why is this example important?**
```
[the remark this answers is not in the document]
Comment (blackfalcon) [reply demo/dotCompliance.md]: Removed content. It's in the commit log if the demo is necessary to be brought back.

File: demo/dotCompliance.md [hunk 6]
```diff
@@ -0,0 +1,376 @@
...
+      </auro-flightline>
+    </auro-flight>
+  ```
+
+</auro-accordion> -->
+
+
+## Mainline + Regional Connection (SEAAVP)
+
+**Why is this example important?**
```
[the remark this answers is not in the document]
Comment (blackfalcon) [reply demo/dotCompliance.md]: Removed content. It's in the commit log if the demo is necessary to be brought back.

**Pull request merged**